src/compiler/scala/tools/nsc/backend/jvm/BytecodeWriters.scala

```diff
@@ -118,10 +135,20 @@ trait BytecodeWriters {
     def writeClass(label: String, jclassName: String, jclassBytes: Array[Byte], outfile: AbstractFile) {
       assert(outfile != null,
              "Precisely this override requires its invoker to hand out a non-null AbstractFile.")
-      val outstream = new DataOutputStream(outfile.bufferedOutput)
+      if (outfile.file != null) {
+        try {
+          Files.write(outfile.file.toPath, jclassBytes)
+        } catch {
+          case _: java.nio.file.NoSuchFileException =>
```

Comment:
I have split out an alternative as https://github.com/scala/scala/pull/5834

will run performance test overnight and post up is successful